tests/scripts/depends.py

```diff
@@ -276,8 +273,10 @@ def test(self, options):
 # These are not necessarily dependencies, but just minimal required changes
 # if a given define is the only one enabled from an exclusive group.
 EXCLUSIVE_GROUPS = {
-    'MBEDTLS_SHA256_C': ['+MBEDTLS_SHA224_C'],
-    'MBEDTLS_SHA384_C': ['+MBEDTLS_SHA512_C'],
+    'MBEDTLS_SHA224_C': ['+MBEDTLS_SSL_COOKIE_C'],
+    'MBEDTLS_SHA256_C': ['-MBEDTLS_SSL_COOKIE_C'],
+    'MBEDTLS_SHA384_C': ['+MBEDTLS_SHA512_C',
+                         '+MBEDTLS_SSL_COOKIE_C'],
```

Comment:
Note: I don't think +COOKIES should be necessary here - it's already part of the full config so we don't need to explicitly enable it. Generally speaking, we should only need "minus" entries here except for special cases like the inter-dependencies that this PR is removing.